WcaOnRails/app/models/competition.rb

```diff
@@ -54,6 +55,13 @@ def must_have_at_least_one_delegate
     end
   end
 
+  validate :delegates_must_be_delegates, on: [:update]
+  def delegates_must_be_delegates
+    if !self.delegates.all? { |d| d.any_kind_of_delegate? }
+      errors.add(:delegate_ids, "You must pass only delegates here")
```

Comment:
Hmmm... "here" feels weird here, how about something like "are not all delegates" instead?